identity/src/keypair.rs

```diff
@@ -108,71 +108,115 @@ impl Keypair {
     }
 
     #[cfg(feature = "ed25519")]
+    #[deprecated(
+        since = "0.2.0",
+        note = "This method name does not follow Rust naming conventions, use `Keypair::try_into_ed25519` instead."
+    )]
     pub fn into_ed25519(self) -> Option<ed25519::Keypair> {
-        #[allow(deprecated)]
-        #[allow(unreachable_patterns)]
-        match self {
-            Keypair::Ed25519(inner) => Some(inner),
-            _ => None,
+        match self.try_into() {
+            Ok(k) => Some(k),
+            Err(_) => None,
         }
```

Comment:
There is a combinator that can do this!

```suggestion
        self.try_into().ok()
```